packages/tldraw/src/lib/TldrawImage.tsx

```diff
@@ -0,0 +1,138 @@
+	useShallowArrayIdentity,
+	useTLStore,
+} from '@tldraw/editor'
+import { useEffect, useLayoutEffect, useMemo, useState } from 'react'
+import { defaultShapeUtils } from './defaultShapeUtils'
+import { usePreloadAssets } from './ui/hooks/usePreloadAssets'
+import { exportToString } from './utils/export/export'
+import { useDefaultEditorAssetsWithOverrides } from './utils/static-assets/assetUrls'
+
+/**
+ * Props for the {@link @tldraw/tldraw#TldrawImage} component.
+ *
+ * @public
+ **/
+export type TldrawImageProps = {
+	/**
+	 * The snapshot to display.
+	 */
+	snapshot: StoreSnapshot<TLRecord>
+
+	/**
+	 * The page to display. Defaults to the first page.
+	 */
+	pageId?: TLPageId
+
+	/**
+	 * Options for the displayed image.
+	 */
+	opts?: Partial<TLSvgOptions>
+
+	/**
+	 * Additional shape utils to use.
+	 */
+	shapeUtils?: readonly TLAnyShapeUtilConstructor[]
+}
+
+/**
+ * A renderered SVG image of a Tldraw snapshot.
+ *
+ * @example
+ * ```tsx
+ * <TldrawImage snapshot={snapshot} />
+ * ```
+ *
+ * @example
+ * ```tsx
+ * <TldrawImage
+ * 	snapshot={snapshot}
+ * 	pageId={pageId}
+ * 	opts={{ background: false, darkMode: true }}
+ * />
+ * ```
+ *
+ * @public
+ */
+export function TldrawImage(props: TldrawImageProps) {
+	const [container, setContainer] = useState<HTMLDivElement | null>(null)
+	const [editor, setEditor] = useState<Editor | null>(null)
+
+	const shapeUtils = useShallowArrayIdentity(props.shapeUtils ?? [])
+	const shapeUtilsWithDefaults = useMemo(() => [...defaultShapeUtils, ...shapeUtils], [shapeUtils])
+	const store = useTLStore({ snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults })
+
+	const assets = useDefaultEditorAssetsWithOverrides()
+	const { done: preloadingComplete, error: preloadingError } = usePreloadAssets(assets)
+
+	useLayoutEffect(() => {
+		if (!container) return
+		const editor = new Editor({
+			store,
+			shapeUtils: shapeUtilsWithDefaults ?? [],
+			tools: [],
+			getContainer: () => container,
+		})
+		setEditor(editor)
+		return () => {
+			editor.dispose()
+		}
+	}, [container, store, shapeUtilsWithDefaults])
+
+	if (preloadingError) {
+		return <ErrorScreen>Could not load assets.</ErrorScreen>
+	}
+
+	if (!preloadingComplete) {
+		return <LoadingScreen>Loading assets...</LoadingScreen>
+	}
+
+	return (
+		<div ref={setContainer} style={{ position: 'relative', width: '100%', height: '100%' }}>
+			{container && (
+				<ContainerProvider container={container}>
+					{editor && (
+						<EditorContext.Provider value={editor}>
+							<TldrawImageSvg pageId={props.pageId} opts={props.opts} />
+						</EditorContext.Provider>
+					)}
+				</ContainerProvider>
+			)}
+		</div>
+	)
+}
+
+function TldrawImageSvg({ pageId, opts }: { pageId?: TLPageId; opts?: Partial<TLSvgOptions> }) {
+	const editor = useEditor()
+	const [url, setUrl] = useState<string | null>(null)
+
+	useEffect(() => {
+		getImageUrl(editor, pageId, opts).then((url) => {
+			setUrl(url)
+		})
+	}, [editor, opts, pageId])
```

Comment:
is this ok?
whats best practice for async useMemo?
or is there a better approach im missing?

Reply:
this is pretty much it! the one thing to add here is a way of stopping the action if/when the thing fails:
```ts
	useEffect(() => {
		let isCancelled = false
		getImageUrl(editor, pageId, opts).then((url) => {
			if (isCancelled) return
			setUrl(url)
		})
		return () => {
			isCancelled = true
		}
	}, [editor, opts, pageId])

Reply:
thanks, done!